Mathlib/CategoryTheory/Preadditive/ProjectiveResolution.lean

```diff
@@ -0,0 +1,365 @@
+  by apply liftHomotopy (𝟙 X) <;> simp
+set_option linter.uppercaseLean3 false in
+#align category_theory.ProjectiveResolution.lift_id_homotopy CategoryTheory.ProjectiveResolution.liftIdHomotopy
+
+/-- The lift of a composition is homotopic to the composition of the lifts. -/
+def liftCompHomotopy {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z) (P : ProjectiveResolution X)
+    (Q : ProjectiveResolution Y) (R : ProjectiveResolution Z) :
+    Homotopy (lift (f ≫ g) P R) (lift f P Q ≫ lift g Q R) := by
+  apply liftHomotopy (f ≫ g) <;> simp
+set_option linter.uppercaseLean3 false in
+#align category_theory.ProjectiveResolution.lift_comp_homotopy CategoryTheory.ProjectiveResolution.liftCompHomotopy
+
+-- We don't care about the actual definitions of these homotopies.
+/-- Any two projective resolutions are homotopy equivalent. -/
+def homotopyEquiv {X : C} (P Q : ProjectiveResolution X) : HomotopyEquiv P.complex Q.complex where
+  hom := lift (𝟙 X) P Q
+  inv := lift (𝟙 X) Q P
+  homotopyHomInvId := by
+    refine' (liftCompHomotopy (𝟙 X) (𝟙 X) P Q P).symm.trans _
+    simp only [Category.id_comp]
+    apply liftIdHomotopy
+  homotopyInvHomId := by
+    refine' (liftCompHomotopy (𝟙 X) (𝟙 X) Q P Q).symm.trans _
+    simp only [Category.id_comp]
+    apply liftIdHomotopy
+set_option linter.uppercaseLean3 false in
+#align category_theory.ProjectiveResolution.homotopy_equiv CategoryTheory.ProjectiveResolution.homotopyEquiv
+
+@[reassoc (attr := simp)]
+theorem homotopyEquiv_hom_π {X : C} (P Q : ProjectiveResolution X) :
+    (homotopyEquiv P Q).hom ≫ Q.π = P.π := by simp [homotopyEquiv]
+set_option linter.uppercaseLean3 false in
+#align category_theory.ProjectiveResolution.homotopy_equiv_hom_π CategoryTheory.ProjectiveResolution.homotopyEquiv_hom_π
+
+@[reassoc (attr := simp)]
+theorem homotopyEquiv_inv_π {X : C} (P Q : ProjectiveResolution X) :
+    (homotopyEquiv P Q).inv ≫ P.π = Q.π := by simp [homotopyEquiv]
+set_option linter.uppercaseLean3 false in
+#align category_theory.ProjectiveResolution.homotopy_equiv_inv_π CategoryTheory.ProjectiveResolution.homotopyEquiv_inv_π
+
+end ProjectiveResolution
+
+section
+
+variable [HasZeroMorphisms C] [HasZeroObject C] [HasEqualizers C] [HasImages C]
+
+def projectiveResolution (Z : C) [HasProjectiveResolution Z] : ProjectiveResolution Z :=
+  HasProjectiveResolution.out.some
+
+-- porting note: this was named `projectiveResolution` is mathlib. As there was also a need
+-- for a definition of `ProjectiveResolution Z` given `(Z : projectiveResolution Z)`, it
+-- seemed more consistent to have `projectiveResolution Z : ProjectiveResolution Z`
+-- and `projectiveResolution.complex Z : ChainComplex C ℕ`
+/-- An arbitrarily chosen projective resolution of an object. -/
+abbrev projectiveResolution.complex (Z : C) [HasProjectiveResolution Z] : ChainComplex C ℕ :=
+  (projectiveResolution Z).complex
+#align category_theory.projective_resolution CategoryTheory.projectiveResolution
```

Comment:
I think this means that the align should be like this, right?
```suggestion
#align category_theory.projective_resolution CategoryTheory.projectiveResolution.complex
```

Reply:
Thanks very much for spotting this!